From 01dc137740b0386036d141a7c1f06290320cbd3a Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Sun, 21 May 2023 01:26:59 +0200
Subject: [PATCH] Input: goodix - try to keep regulator enable/disable balanced

Failure handling is still not optimal in this driver, but let's
try to improve it.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 drivers/input/touchscreen/goodix.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/input/touchscreen/goodix.c b/drivers/input/touchscreen/goodix.c
index 7578ca32a953..24218440a7e6 100644
--- a/drivers/input/touchscreen/goodix.c
+++ b/drivers/input/touchscreen/goodix.c
@@ -1516,6 +1516,7 @@ static int goodix_resume(struct device *dev)
 
 		error = regulator_enable(ts->vddio);
 		if (error) {
+			regulator_disable(ts->avdd28);
 			dev_err(dev, "Regulator vddio enable failed.\n");
 			return error;
 		}
-- 
2.43.0

